strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/services/impl/ArtifactEntryServiceImpl.java

```diff
@@ -1,91 +1,83 @@
-import org.springframework.transaction.annotation.Propagation;
 import org.springframework.transaction.annotation.Transactional;
 
+import com.orientechnologies.orient.core.sql.query.OSQLSynchQuery;
+
 /**
  * DAO implementation for {@link ArtifactEntry} entities.
  *
  * @author Alex Oreshkevich
  */
 @Service
 @Transactional
-class ArtifactEntryServiceImpl
-        implements ArtifactEntryService
+class ArtifactEntryServiceImpl extends CommonCrudService<ArtifactEntry> implements ArtifactEntryService
 {
 
     private static final Logger logger = LoggerFactory.getLogger(ArtifactEntryService.class);
 
     // will help us avoid to have hardcoded name of this class
     private static final String ARTIFACT_ENTRY_CLASS_NAME = ArtifactEntry.class.getSimpleName();
 
-    @Inject
-    ArtifactRepository artifactRepository;
-
-    @Inject
-    CacheManager cacheManager;
-
-    @Inject
-    OObjectDatabaseTx databaseTx;
+    @Override
+    public Class<ArtifactEntry> getEntityClass()
+    {
+        return ArtifactEntry.class;
+    }
 
+    @PersistenceContext
+    private EntityManager entityManager;
+    
     @Override
-    @SuppressWarnings("unchecked")
-    // don't try to use second level cache here until you make all coordinates properly serializable
-    public List<ArtifactEntry> findByCoordinates(ArtifactCoordinates coordinates)
+    @Transactional
+    public List<ArtifactEntry> findByCoordinates(Map<String, String> coordinates)
     {
-        // if search query is null or empty delegate to #findAll
-        if (coordinates == null || coordinates.getCoordinates()
-                                              .keySet()
+        if (coordinates == null || coordinates.keySet()
                                               .isEmpty())
         {
             return findAll().orElse(Collections.EMPTY_LIST);
         }
 
         // prepare custom query based on all non-null coordinates that were joined by logical AND
         // read more about fetching strategies here: http://orientdb.com/docs/2.2/Fetching-Strategies.html
-        String nativeQuery = buildQueryFrom(coordinates);
-        OQueryAbstract query = new OSQLSynchQuery<>(nativeQuery).setFetchPlan("*:-1");
 
+        String nativeQuery = buildQuery(coordinates);
+        OSQLSynchQuery<ArtifactEntry> query = new OSQLSynchQuery<>(nativeQuery);
         logger.info("[findByCoordinates] SQL -> \n\t" + nativeQuery);
 
-        databaseTx.activateOnCurrentThread();
-        List<ArtifactEntry> resultList = databaseTx.query(query);
-
-        // still have to detach everything manually until we fully migrate to OrientDB 2.2.X
-        // where fetching strategies will be fully supported
-        List<ArtifactEntry> detachedList = new LinkedList<>();
-        resultList.forEach(artifactEntry -> detachedList.add(databaseTx.detachAll(artifactEntry, true)));
+        List<ArtifactEntry> resultList = getDelegate().query(query);
+        return resultList;
```

Comment:
@sbespalov : Unless you intended this for debugging purposes, I think it sounds like a reasonable suggestion.